src/averageAges.js

```diff
@@ -15,11 +15,14 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const choiceMan = people.filter(person => (
+    century
+      ? person.sex === 'm' && Math.ceil(person.died / 100) === century
+      : person.sex === 'm'),
+  );
+
+  return choiceMan.reduce((x = 0, { born, died }) =>
+    x + (died - born), 0) / choiceMan.length;
```

Comment:
better to set the result of reduce to a variable, now it a little difficult to read